api/v1/contexts/PKPContextHandler.inc.php

```diff
@@ -0,0 +1,490 @@
+		$themePluginPath = empty($params['themePluginPath']) ? null : $params['themePluginPath'];
+		if ($themePluginPath !== $context->getData('themePluginPath')) {
+			$errors = $contextService->validate(
+				VALIDATE_ACTION_EDIT,
+				['themePluginPath' => $themePluginPath],
+				$context->getSupportedLocales(),
+				$context->getPrimaryLocale()
+			);
+			if (!empty($errors)) {
+				return $response->withJson($errors, 400);
+			}
+			$newContext = $contextService->editContext($context, ['themePluginPath' => $themePluginPath], $request);
+		}
+
+		// Get the appropriate theme plugin
+		$allThemes = PluginRegistry::loadCategory('themes', true);
+		$selectedTheme = null;
+		foreach ($allThemes as $theme) {
+			if ($themePluginPath === $theme->getDirName()) {
+				$selectedTheme = $theme;
+				break;
+			}
+		}
+
+		// Run the theme's init() method if a new theme has been selected
+		if (isset($newContext)) {
+			$selectedTheme->init();
+		}
+
+		$errors = $selectedTheme->validateOptions($params, $themePluginPath, $context->getId(), $request);
+		if (!empty($errors)) {
+			return $response->withJson($errors, 400);
+		}
+
+		// Only accept params that are defined in the theme options
+		$options = $selectedTheme->getOptionsConfig();
+		foreach ($options as $optionName => $optionConfig) {
+			if (!array_key_exists($optionName, $params)) {
+				continue;
+			}
+			$selectedTheme->saveOption($optionName, $params[$optionName], $context->getId());
+		}
+
+		// Clear the template cache so that new settings can take effect
+		$templateMgr = TemplateManager::getManager(Application::getRequest());
+		$templateMgr->clearTemplateCache();
+		$templateMgr->clearCssCache();
+
+		$data = array_merge(
+			$selectedTheme->getOptionValues($context->getId()),
+			['themePluginPath' => $themePluginPath]
+		);
+
+		ksort($data);
+
+		return $response->withJson($data, 200);
+	}
+
+	/**
+	 * Delete a context
+	 * @param $slimRequest Request Slim request object
+	 * @param $response Response object
+	 * @param array $args arguments
+	 *
+	 * @return Response
+	 */
+	public function deleteContext($slimRequest, $response, $args) {
+
+		// This endpoint is only available at the site-wide level
+		if ($this->getRequest()->getContext()) {
+			return $response->withStatus(404)->withJsonError('api.submissions.404.siteWideEndpoint');
+		}
+
+		$userRoles = $this->getAuthorizedContextObject(ASSOC_TYPE_USER_ROLES);
+		if (!in_array(ROLE_ID_SITE_ADMIN, $userRoles)) {
+			$response->withStatus(403)->withJsonError('api.contexts.403.notAllowedDelete');
+		}
+
+		$contextId = (int) $args['contextId'];
+
+		$contextService = ServicesContainer::instance()->get('context');
+		$context = $contextService->getContext($contextId);
+
+		if (!$context) {
+			return $response->withStatus(404)->withJsonError('api.contexts.404.contextNotFound');
+		}
+
+		$contextProps = $contextService->getSummaryProperties($context, array(
+				'request' => $this->getRequest(),
```

Comment:
(Looks like this is indented more than usual)

Reply:
Good catch!